src/async_fuse/memfs/s3_node.rs

```diff
@@ -62,17 +57,15 @@ impl S3NodeData {
                 }
                 SerialNodeData::Directory(serial_dir)
             }
-            Self::RegFile(_) => SerialNodeData::File,
+            Self::RegFile => SerialNodeData::File,
             Self::SymLink(ref target) => SerialNodeData::SymLink(target.clone()),
         }
     }
 }
 
 /// A file node or a directory node
 #[derive(Debug)]
-pub struct S3Node<S: S3BackEnd + Sync + Send + 'static> {
-    /// S3 Backend
-    s3_backend: Arc<S>,
+pub struct S3Node {
```

Comment:
We should rename `S3Metadata` and `S3Node` since they have nothing to do with "S3".

Reply:
Make sense. And the trait of `Metadata` and `Node` should be considered further.